registrations/api.py

```diff
@@ -157,7 +157,15 @@ def destroy(self, request, *args, **kwargs):
 
     def filter_queryset(self, queryset):
         events = Event.objects.exclude(registration=None)
-        events = _filter_event_queryset(events, self.request.query_params)
+
+        query_params = self.request.query_params.copy()
+        # By default don't filter events by event_type
```

Comment:
Indeed it seems that without this, you would only get events with `GENERAL` type from `_filter_event_queryset`. Maybe something about that could be added to the comment just to document this part for later generations.

Reply:
> Also, does the copy serve some purpose?

The purpose is to get mutable version of query_params. `query_params["event_type"] = ",".join(event_types.keys())` throws **AttributeError: This QueryDict instance is immutable** error without the copy.
Added a comment and hopefully it makes more sense now.

Other option would be set query_params mutable explicitly: `query_params._mutable=True`

Reply:
> Indeed it seems that without this, you would only get events with `GENERAL` type from `_filter_event_queryset`. Maybe something about that could be added to the comment just to document this part for later generations.

Added a comment to clarify the reason to add event_type attribute if not set.

Reply:
Oh I see, didn't realize it's immutable. I think copy is better than changing _mutable. :+1: